### PR TITLE
Oct 2655 bt stream module

### PR DESCRIPTION
**Pull request by alexsven**

### CODE REVIEW

File: applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_source.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ * @return	0 for success, error otherwise.
+ */
+int broadcast_source_send(struct encoded_audio enc_audio);
+
+/**
+ * @brief	Enable the LE Audio broadcast (BIS) source.
+ *
+ * @return	0 for success, error otherwise.
+ */
+int broadcast_source_enable(void);
```
Comment (alexsven) [inline applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_source.h:55]: Remove. It seems this got messed up in a rebase in https://github.com/alexsven/sdk-nrf/pull/25/files#diff-f960e1ec977a994f52a902c0645a161e09aae73315c58ac9604efa66a04ca6f4R51-R56

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c [hunk 2]
```diff
@@ -549,182 +753,3 @@ static int initialize(le_audio_receive_cb recv_cb, le_audio_timestamp_cb timestm
 
 	return 0;
 }
```
Comment (alexsven) [inline applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c:755]: There should be a unicast_server_disable

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.h [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+			      uint32_t *pres_delay_us);
+
+/**
+ * @brief	Get advertising data from this module.
+ *
+ * @note	This partial data is used to build a complete extended advertising packet.
+ *
+ * @param [in/out] uuid_buf	Buffer being populated with UUIDs.
+ * @param [in/out] adv_buf	Buffer being populated with ext adv elements.
+ * @param [in] adv_buf_vacant	Number of vacant elements in @p adv_buf.
```
Comment (alexsven) [inline applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.h:44]: Formatting not correct

File: applications/nrf5340_audio/src/bluetooth/Kconfig [hunk 4]
```diff
@@ -13,12 +13,24 @@ rsource "bt_management/Kconfig"
...
 
+config EXT_ADV_BUF_MAX
+	int "Max number of extended advertising data parameters"
+	default 10
+
+config EXT_ADV_UUID_BUF_MAX
+	int "Max number of UUIDs to add to the ext adv"
+	default 10
+
 choice BT_BAP_UNICAST_BAP_CONFIGURATION
```
Comment (alexsven) [inline applications/nrf5340_audio/src/bluetooth/Kconfig:34]: All of the stuff below here should be moved to bt_stream/unicast/Kconfig

File: applications/nrf5340_audio/src/bluetooth/Kconfig.defaults [hunk 5]
```diff
@@ -16,9 +16,6 @@ config BT_ECC
 config BT_EXT_ADV
 	default y
 
-config BT_DEVICE_NAME
-	default "NRF5340_AUDIO"
-
 # Mandatory to support at least 1 for ASCS
 config BT_ATT_PREPARE_COUNT
 	default 1
```
Comment (alexsven) [inline applications/nrf5340_audio/src/bluetooth/Kconfig.defaults:21]: Code below related to unicast should be moved to bt_stream/unicast/Kconfig.default

**Review by alexsven — COMMENTED** [review]
Initial comments based on reviewing the two last commits only, since I didn't get to review those PRs last week

File: applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_sink.c [hunk 6]
```diff
@@ -612,51 +566,41 @@ int le_audio_pause(void)
...
 	}
 
-	LOG_DBG("LE Audio enabled");
+	LOG_DBG("Broadcast sink disabled");
 
 	return 0;
 }
 
-int le_audio_disable(void)
+int broadcast_sink_enable(le_audio_receive_cb recv_cb)
```
Comment (alexsven) [inline applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_sink.c:593]: The only thing this function does now is call initialize, so I suggest we just move the code from initialize here.

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c [hunk 7]
```diff
@@ -53,20 +54,51 @@ enum csip_set_rank {
...
+
+static const uint8_t cap_adv_data[] = {
+	BT_UUID_16_ENCODE(BT_UUID_CAS_VAL),
+	BT_AUDIO_UNICAST_ANNOUNCEMENT_TARGETED,
+};
+
+static uint8_t tmap_adv_data[] = {
+	BT_UUID_16_ENCODE(BT_UUID_TMAS_VAL),
+	BT_BYTES_LIST_LE16(BT_TMAP_ROLE_UMR | BT_TMAP_ROLE_CT),
+};
```
[the remark this answers is not in the document]
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c:78]: I will remove in my testing branch

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c [hunk 8]
```diff
@@ -13,6 +13,7 @@
 #include <zephyr/bluetooth/audio/pacs.h>
 #include <zephyr/bluetooth/audio/csip.h>
 #include <zephyr/bluetooth/audio/cap.h>
+#include <zephyr/bluetooth/audio/tmap.h>
```
[the remark this answers is not in the document]
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c:16]: I will remove in my testing branch

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c [hunk 9]
```diff
@@ -53,20 +54,51 @@ enum csip_set_rank {
...
+#define AVAILABLE_SINK_CONTEXT (BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED | BT_AUDIO_CONTEXT_TYPE_MEDIA)
+#else
+#define AVAILABLE_SINK_CONTEXT 0x0000
+#endif
+
+static uint8_t unicast_server_adv_data[] = {
+	BT_UUID_16_ENCODE(BT_UUID_ASCS_VAL),
+	BT_AUDIO_UNICAST_ANNOUNCEMENT_TARGETED,
+	BT_BYTES_LIST_LE16(AVAILABLE_SINK_CONTEXT),
+	BT_BYTES_LIST_LE16(AVAILABLE_SOURCE_CONTEXT),
```
[the remark this answers is not in the document]
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c:100]: I will do this change in my test branch

**Comment by greg-fer** [comment]
Discussed this with @erikrobstad , and when the PR is open and not in draft anymore, I'll work on the following updates:
- Review of the API docs in header files.
- Update to https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/applications/nrf5340_audio/doc/firmware_architecture.html with new information about BT modules.

**Comment by greg-fer** [comment]
@alexsven , my PR is merged. Please remember to pull edits. Thanks.

**Review by greg-fer — APPROVED** [review]

**Comment by erikrobstad** [comment]
**Remember that the commit message must be changed before merging it to NCS main**

**Review by erikrobstad — CHANGES_REQUESTED** [review]
Commit message must be changed

**Review by erikrobstad — COMMENTED** [review]
Never mind

**Review by koffes — CHANGES_REQUESTED** [review]

**Review by koffes — APPROVED** [review]

**Review by erikrobstad — APPROVED** [review]

**Review by Frodevan — APPROVED** [review]

**Pull request merged**